### PR TITLE
Chore/#71 refactor tests using msw

### PR DESCRIPTION
**Pull request by kylejwatson**
# Description

Added msw to avoid mocking services and test UI closer to real usage.
Adds tests for GroupDataService so that we have full coverage.

### Issue

This PR addresses issue #71.

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/DevoteamNL/opendora/blob/main/CONTRIBUTING.md).**
- [x] **I have ran the linting and formatting scripts to make sure the code is consistently styled.**
- [x] **I have built/compiled and ran the tests to make sure the code works correctly.**
- [x] **I have added or changed tests to cover new or different functionality, where applicable.**
- [x] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### CODE REVIEW

File: backstage-plugin/plugins/open-dora/src/services/GroupDataService.test.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      await service.retrieveMetricDataPoints({
+        type: 'df_count',
+        aggregation: 'monthly',
+        project: 'project1',
+        team: 'team1',
+      }),
+    ).toEqual({
+      aggregation: 'monthly',
+      dataPoints: [
+        { key: 'project1_team1_monthly_df_count_first_key', value: 1 },
```
Comment (duke-b) [inline backstage-plugin/plugins/open-dora/src/services/GroupDataService.test.ts:61]: I assume it wont cause any issues 1.0 (from line 22) and just 1

**Review by duke-b — DISMISSED** [review]
Looks good

File: backstage-plugin/plugins/open-dora/src/services/GroupDataService.test.ts (hunk 1 above)
Comment (kylejwatson) [reply backstage-plugin/plugins/open-dora/src/services/GroupDataService.test.ts:61]: Good spot, don't think so though since `1 === 1.0` is truth

Comment (kylejwatson) [reply backstage-plugin/plugins/open-dora/src/services/GroupDataService.test.ts:61]: I can change it though just so the tests are clearer

**Review by duke-b — APPROVED** [review]
Looks good

**Pull request merged**